core/math/vector3i.h

```diff
@@ -110,6 +119,35 @@ struct Vector3i {
 	}
 };
 
+Vector3i Vector3i::cross(const Vector3i &p_with) const {
+	Vector3i ret(
+			(y * p_with.z) - (z * p_with.y),
+			(z * p_with.x) - (x * p_with.z),
+			(x * p_with.y) - (y * p_with.x));
+
+	return ret;
+}
+
+int64_t Vector3i::dot(const Vector3i &p_with) const {
+	return x * int64_t(p_with.x) + y * int64_t(p_with.y) + z * int64_t(p_with.z);
+}
+
+double Vector3i::length() const {
+	int64_t x2 = x * int64_t(x);
+	int64_t y2 = y * int64_t(y);
+	int64_t z2 = z * int64_t(z);
+
+	return Math::sqrt(double(x2 + y2 + z2));
+}
```

Comment:
Could be `sqrt` of `length_squared()` I guess unless we're really concerned that it wouldn't be inlined?

Reply:
It was done this way to be consistent with how it's currently done in Vector3, but I agree that it's simpler to just take the `sqrt` of `length_squared()`, so updated. I left Vector3 alone.